### PR TITLE
ReactJS warnings resolved in test suite

### PR DESCRIPTION
**Pull request by bmallred**
Issue #443 

## TODO

 - [x] Warning: Each child in an array or iterator should have a unique "key" prop.

### CODE REVIEW

File: src/components/Form/BranchCollection/BranchCollection.jsx [hunk 1]
```diff
@@ -126,43 +135,52 @@ export default class BranchCollection extends React.Component {
...
+    const bottom = (index, item) => {
+      return this.props.items.length - 1 === index
+        ? this.branch({
+          className: 'last-branch',
+          onUpdate: this.onLastBranchClick.bind(this)
+        })
+      : null
+    }
+
+    const kiddos = (index, item) => {
```
Comment (floresj) [inline src/components/Form/BranchCollection/BranchCollection.jsx]: :+1: variable name

File: src/components/Form/Collection/Collection.jsx [hunk 2]
```diff
@@ -8,14 +8,19 @@ export default class Collection extends ValidationElement {
   constructor (props) {
     super(props)
 
-    let min = this.props.minimum || 0
+    let indices = []
+    for (let i = 0; i < this.props.items.length; i++) {
+      indices.push(super.guid())
+    }
+    const f = this.factory(this.props.minimum, this.props.items, indices)
```
Comment (adelevie) [inline src/components/Form/Collection/Collection.jsx]: what is `f`?

File: src/components/Form/Help/Help.jsx [hunk 3]
```diff
@@ -143,27 +143,31 @@ export default class Help extends ValidationElement {
...
       let extendedProps = {}
 
-      if (child.type) {
-        let what = Object.prototype.toString.call(child.type)
-        if (what === '[object Function]' && child.type.name === 'HelpIcon') {
-          extendedProps.onClick = this.handleClick
-          extendedProps.active = this.state.active
+      if (React.isValidElement(child)) {
+        if (child.type) {
+          const what = Object.prototype.toString.call(child.type)
```
Comment (adelevie) [inline src/components/Form/Help/Help.jsx]: what is `what`?

**Comment by adelevie** [comment]
LGTM

File: src/components/Form/Collection/Collection.jsx (hunk 2 above)
Comment (bmallred) [inline src/components/Form/Collection/Collection.jsx]: A factory.

**Pull request merged**